src/app/_shared-components/Preimages/UserPreimagesTab/UserPreimagesTab.tsx

```diff
@@ -0,0 +1,160 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+'use client';
+
+import React from 'react';
+import { ERROR_CODES, ERROR_MESSAGES } from '@/_shared/_constants/errorLiterals';
+import { NextApiClientService } from '@/app/_client-services/next_api_client_service';
+import { ClientError } from '@/app/_client-utils/clientError';
+import { useUser } from '@/hooks/useUser';
+import { useUserPreferences } from '@/hooks/useUserPreferences';
+import { useSearchParams } from 'next/navigation';
+import { useQuery } from '@tanstack/react-query';
+import { useTranslations } from 'next-intl';
+import { Button } from '@ui/Button';
+import Link from 'next/link';
+import styles from '@ui/Preimages/SearchBar/SearchBar.module.scss';
+import { RefreshCcwIcon } from 'lucide-react';
+import ListingTable from '../ListingTable/ListingTable';
+import NoUserPreimage from '../ListingTable/NoUserPreimage';
+import SearchBar from '../SearchBar/SearchBar';
+import SwitchWalletOrAddress from '../../SwitchWalletOrAddress/SwitchWalletOrAddress';
+import Address from '../../Profile/Address/Address';
+import { Dialog, DialogContent, DialogHeader, DialogTitle, DialogTrigger } from '../../Dialog/Dialog';
+
+function UserPreimagesTab() {
+	const searchParams = useSearchParams();
+	const page = parseInt(searchParams?.get('page') || '1', 10);
+	const { user } = useUser();
+	const { userPreferences } = useUserPreferences();
+	const t = useTranslations('Preimages');
+
+	const selectedAddress = userPreferences?.selectedAccount?.address || user?.addresses?.[0] || '';
+
+	const {
+		data: userPreimagesData,
+		isLoading,
+		error,
+		refetch
+	} = useQuery({
+		queryKey: ['userPreimages', selectedAddress, page, user?.id, userPreferences?.selectedAccount?.address],
+		queryFn: async () => {
+			if (!selectedAddress) {
+				return null;
+			}
+
+			const { data, error: fetchError } = await NextApiClientService.fetchUserPreimages({
+				page,
+				address: selectedAddress
+			});
+
+			if (fetchError || !data) {
+				throw new ClientError(ERROR_CODES.CLIENT_ERROR, fetchError?.message || ERROR_MESSAGES[ERROR_CODES.CLIENT_ERROR]);
+			}
+
+			return data;
+		},
+		enabled: !!selectedAddress && !!user?.addresses?.length,
+		retry: 1
+	});
+
+	// Show loading state
+	if (isLoading) {
```

Comment:
don't replace the whole component with loader, just show loader on the table, use masked loader, LoadingLayover component already there. please take reference